medusa/providers/torrent/html/shanaproject.py

```diff
@@ -0,0 +1,186 @@
+# coding=utf-8
+
+"""Provider code for ShanaProject."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+import traceback
+
+from dateutil import parser
+
+from medusa import tv
+from medusa.bs4_parser import BS4Parser
+from medusa.helper.common import convert_size
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class ShanaProjectProvider(TorrentProvider):
+    """ShanaProject Torrent provider."""
+
+    def __init__(self):
+        """Initialize the class."""
+        super(self.__class__, self).__init__('ShanaProject')
```

Comment:
Need to avoid copy-pasta errors.  The providers should never have had this change merged as its breaks the init.  When resolving the super call it starts at the subclass instead of the intended starting position causing an infinite recursion error.  Proper super would be `super(MyClass, self).__init__()`.